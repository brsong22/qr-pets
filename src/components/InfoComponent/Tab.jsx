import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Tab extends Component {
  constructor(props) {
    super(props);
    this.handleTabClick = this.handleTabClick.bind(this);
  }

  handleTabClick(event) {
    const { onClick, tabText } = this.props;
    onClick(event, tabText);
  }

  render() {
    const { tabText, selected } = this.props;
    const show = selected ? 'selected' : '';
    return (
      <button className={`tab ${show}`} id={`#${tabText}_tab`} type="button" onClick={this.handleTabClick}>
        {tabText}
      </button>
    );
  }
}

Tab.propTypes = {
  tabText: PropTypes.string,
  selected: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
};

Tab.defaultProps = {
  tabText: '',
  selected: false,
};

export default Tab;
