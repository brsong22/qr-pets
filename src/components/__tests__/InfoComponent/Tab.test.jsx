
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Tab from '../../InfoComponent/Tab';

describe('Tab', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<Tab />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
