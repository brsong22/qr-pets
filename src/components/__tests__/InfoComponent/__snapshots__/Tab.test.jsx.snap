// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab renders as expected 1`] = `
<button
  className="tab "
  id="#_tab"
  onClick={[Function]}
  type="button"
/>
`;
